packages/pigeon/platform_tests/ios_swift_unit_tests/ios/RunnerTests/HandlerBinaryMessenger.swift

```diff
@@ -0,0 +1,43 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import Foundation
+import Flutter
+
+typealias HandlerBinaryMessengerHandler = ([Any?])-> Any
+
+class HandlerBinaryMessenger: NSObject, FlutterBinaryMessenger {
+  let codec: FlutterMessageCodec
+  let handler: HandlerBinaryMessengerHandler
+  private var count = 0
+  
+  init(codec: FlutterMessageCodec, handler: @escaping HandlerBinaryMessengerHandler) {
+    self.codec = codec
+    self.handler = handler
+    super.init()
+  }
+  
+  func send(onChannel channel: String, message: Data?) {
+    // Method not implemented because this messenger is just for handling
+  }
+  
+  func send(onChannel channel: String, message: Data?, binaryReply callback: FlutterBinaryReply? = nil) {
+    guard let args = self.codec.decode(message) as? [Any?] else {
+      callback?(nil)
+      return
+    }
+    
+    let result = self.handler(args)
+    callback?(self.codec.encode(result))
```

Comment:
i would do a null check for callback as the first step and skip the whole function if its nil

Reply:
Done